server/driver_tidb.go

```diff
@@ -391,7 +391,7 @@ func convertColumnInfo(fld *ast.ResultField) (ci *ColumnInfo) {
 			// Consider the decimal point.
 			ci.ColumnLength++
 		}
-	} else if fld.Column.Tp != mysql.TypeBit && fld.Column.Tp != mysql.TypeTiny {
+	} else if fld.Column.Tp != mysql.TypeBit && fld.Column.Tp != mysql.TypeTiny && types.IsString(fld.Column.Tp) {
```

Comment:
I think we can only use `types.IsString(fld.Column.Tp)`.
Could we test the cases in the comments at line 395 to 399, maybe we need a Navicat.

Reply:
I will have a try.

Reply:
In PR's description, you say this fixes too many leading 0 before `year`. I am confused why you check column type is `string` or not.

Reply:
@zhexuany , according to the comments below, this else branch is only used for the `show create table` results when it meets Navicat. I think adding a string type check is suitable here.
You could blame the history of this line, it caused at least 2 issues in the past.

Reply:
@imtbkcat 
The `types.IsString` covers the previous conditions, we can remove them.

Reply:
@coocood 
Get it.

Reply:
"according to the comments below, this else branch is only used for the show create table results when it meets Navicat." I had read this part and understand this change made to make navicat happy. While the problem stil remains open: why do we only check the column type is string or not? The reason I ask is to clarify this. And I think it is worth to be commented.